introduction.md

```diff
@@ -26,7 +26,108 @@ CircleCI runs **all types of software tests** including web, mobile, and contain
 **Caching of requirements installation and third-party dependencies.** Instead of installing the environments, CircleCI can take data from multiple projects using the granular check-out-key options.
 **No need for manual debugging.** CircleCI has debugging feature Debug via SSH, but Jenkins users have to debug manually by clicking on jobs.
 
-## Images(System + Services)
+## Executors and Images
+
+#### Overview
+
+CircleCI enables you to run jobs in one of three environments:
+
+* Within Docker images (docker)
+* Within a Linux virtual machine (VM) image (machine)
+* Within a macOS VM image (macos)
+
+It is possible to specify a different executor type for every job in your `.circleci/config.yml` by specifying the 
+executor type and an appropriate image.
+Main differences between using Docker and VM:
+
+| Virtual Environment                        | Docker  | Machine   |
+|--------------------------------------------|---------|-----------|
+| Start time                                 | Instant | 30-60 sec |
+| Clean environment                          | Yes     | Yes       |
+| Custom images                              | Yes     | No        |
+| Build Docker images                        | Yes     | Yes       |
+| Full control over job environment          | No      | Yes       |
+| Full root access                           | No      | Yes       |
+| Run multiple databases                     | Yes     | Yes       |
+| Run multiple versions of the same software | No      | Yes       |
+| Layer caching                              | Yes     | Yes       |
+| Run privileged containers                  | No      | Yes       |
+| Use docker compose with volumes            | No      | Yes       |
+| Configurable resources (CPU/RAM)           | Yes     | No        |
+
+The docker key defines Docker as the underlying technology to run your jobs using Docker Containers. 
+Containers are an instance of the Docker Image you specify. The first image listed in your configuration is 
+the primary container image in which all steps run. All other containers run in a common network and every exposed 
+port will be available on localhost from a primary container.
+
+#### Pre-Built CircleCI Docker Images
+
+For convenience, CircleCI maintains several Docker images. These images are typically extensions of official Docker 
+images and include tools especially useful for CI/CD. All of these pre-built images are available in the [CircleCI org
+on Docker Hub](https://hub.docker.com/search?q=circleci&type=image). 
+
+CircleCI’s convenience images fall into two categories: **language images** and **service images**. 
+All images add a circleci user as a system user.
+
+##### Language Images
```

Comment:
should the font be smaller than in **Pre-Built CircleCI Docker Images** heading? maybe it should be `####`?